lib/streamlit/config.py

```diff
@@ -296,6 +293,56 @@ def _global_unit_test():
     type_=int,
 )
 
+
+# Config Section: Logger #
+_create_section("logger", "Settings to customize Streamlit log messages.")
+
+
+@_create_option("logger.datetimeFormat", type_=str)
+def _logger_datetime_format():
+    """Datetime format for date in logger message. Providing a datetime format
+    will automatically add a timestamp to all logger messages.
+
+    Default: None
+    """
+    if get_option("global.developmentMode"):
+        return "%Y-%m-%dT%H:%M:%SZ"
+    else:
+        return None
+
+
+@_create_option("logger.logLevel", type_=str)
+def _logger_log_level():
+    """Level of logging: 'error', 'warning', 'info', or 'debug'.
+
+    Default: 'info'
+    """
+
+    if get_option("global.logLevel"):
+        return get_option("global.logLevel")
+    elif get_option("global.developmentMode"):
+        return "debug"
+    else:
+        return "info"
+
+
+@_create_option("logger.messageFormat", type_=str)
+def _logger_message_format():
```

Comment:
This also feels like overconfiguration. Why is this needed?

Reply:
It was requested in the issue but it could definitely be overconfiguration. I wasn't sure how much control to give users so I went with more control since it was requested

> I would like to be able to either
>
> run my own configuration script (including the above) on server start up
> or
>
> set a configuration setting in the config.toml file like
> loggingFormat="%(asctime)s %(name)s: %(message)s"